### PR TITLE
[MRG] Fix docs and numpydoc

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Closes: #38 

Adds numpydoc to circleCI sphinx build and fixes numpydoc errors/warnings.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-connectivity/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_connectivity/effective.py [hunk 1]
```diff
@@ -5,14 +5,15 @@
...
+from mne.utils import logger
 
+from .utils import fill_doc
 from .base import SpectralConnectivity, SpectroTemporalConnectivity
 from .spectral import spectral_connectivity
 
 
-@verbose
-def phase_slope_index(data, indices=None, names=None, sfreq=2 * np.pi,
+@fill_doc
```
Comment (larsoner) [inline mne_connectivity/effective.py:15]: This has a `verbose` kwarg so this should stay `verbose`:
```suggestion
@verbose
@fill_doc
```
The `@verbose` kwarg does two things: fills doc, then also sets the `logger.level` if `verbose is not None`. If you change this to just `fill_doc`, you lose the second part. By doing `@fill_doc` then `@verbose`, you should get both.

**Comment by larsoner** [comment]
CircleCI failure is fixed by #36. I fixed a conflict there, will then merge that and rerun here after fixing the `verbose` stuff I mention above

File: mne_connectivity/effective.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 from .spectral import spectral_connectivity
 
 
+@verbose
```
Comment (larsoner) [inline mne_connectivity/effective.py:15]: Looks like you're missing an import for this. There was also more than one place that this is needed, I only pointed out one example, check the diff

**Comment by adam2392** [comment]
I'm not sure why but adding the verbose decorator from MNE and then the fill_doc from MNE-connectivity gives these additional weird errors that are not fixable?

```
reading sources... [100%] whats_new_previous_releases                                                                              
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_connectivity.phase_slope_index':
  PR09: Parameter "mt_low_bias" description should finish with "."

WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_connectivity.phase_slope_index':
  PR09: Parameter "mt_low_bias" description should finish with "."

/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: Inline substitution_reference start-string without end-string.
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: Inline substitution_reference start-string without end-string.
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: Inline substitution_reference start-string without end-string.
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: Inline substitution_reference start-string without end-string.
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: Inline substitution_reference start-string without end-string.
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: Inline substitution_reference start-string without end-string.
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: Inline substitution_reference start-string without end-string.
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: Inline substitution_reference start-string without end-string.
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: Inline substitution_reference start-string without end-string.
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] whats_new_previous_releases                                                                               
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: py:obj reference target not found: mne.stats.{}
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: py:obj reference target not found: EpochsTFR
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: py:obj reference target not found: AverageTFR
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: py:func reference target not found: eeglabio.utils.cart_to_eeglab
/Users/adam2392/Documents/mne-connectivity/mne_connectivity/effective.py:docstring of mne_connectivity.effective.phase_slope_index:83: WARNING: py:func reference target not found: dipy.align._public.affine_registration
```

**Comment by larsoner** [comment]
Some of those are `%`-related issues, others are `intersphinx`. Will push a commit...

**Comment by adam2392** [comment]
Thanks @larsoner for the catches

**Pull request merged**